### PR TITLE
Add Testable instead of Target in TestAction

### PR DESCRIPTION
**Pull request by rowwingman**
Resolves https://github.com/tuist/tuist/issues/707

### Short description 📝

`Sheme.TestAction` replace targets with `testableTargets`. New type contains target name, and tests information how to execute target's tests.

### Solution 📦

Provide broking change by replacing `TestAction.targets` with `TestAction.testableTargets`

### Implementation 👩‍💻👨‍💻

Provide new struct that describes tests info for target, `TestableTarget`.

### CODE REVIEW

File: Sources/ProjectDescription/Scheme.swift [hunk 1]
```diff
@@ -112,6 +112,26 @@ public struct TestAction: Equatable, Codable {
...
+
+    public init(target: String, skipped: Bool = false, parallelizable: Bool = false, randomExecutionOrdering: Bool = false) {
+        self.target = target
+        self.isSkipped = skipped
+        self.isParallelizable = parallelizable
+        self.isRandomExecutionOrdering = randomExecutionOrdering
+    }
+}
+
+extension TestableTarget: ExpressibleByStringLiteral {
```
Comment (pepibumur) [inline Sources/ProjectDescription/Scheme.swift]: Nitpick: What about defining the conformance in the definition of the struct:

```swift
public struct TestableTarget: Equatable, Hashable, Codable, ExpressibleByStringLiteral {
// Content
}
```

File: Sources/TuistCore/Models/Scheme.swift [hunk 2]
```diff
@@ -153,6 +153,26 @@ public class TestAction: Equatable {
     }
 }
 
+public struct TestableTarget: Equatable, Hashable {
```
Comment (pepibumur) [inline Sources/TuistCore/Models/Scheme.swift]: Similar comment here in regards to defining the conformance of `ExpressibleByStringLiteral` here.

**Comment by pepibumur** [comment]
[This](https://help.github.com/en/actions/automating-your-workflow-with-github-actions/events-that-trigger-workflows#pull-request-events-for-forked-repositories) might explain why actions are not ran. Would you mind double checking it @rowwingman ?

**Comment by pepibumur** [comment]
[This](https://help.github.com/en/actions/automating-your-workflow-with-github-actions/events-that-trigger-workflows#pull-request-events-for-forked-repositories) might explain why actions are not ran. Would you mind double checking it @rowwingman ?

**Comment by rowwingman** [comment]
@pepibumur have no idea, what is wrong, enabled actions, so should work fine, but not

**Pull request closed without merging**